ggml/src/ggml-vulkan/vulkan-shaders/mul_mat_vec_q6_k.comp

```diff
@@ -21,21 +23,19 @@ void main() {
     const uint num_blocks_per_row = p.ncols / QUANT_K;
     const uint ib0 = a_offset / QUANT_K + row*num_blocks_per_row;
 
-    const uint tid = gl_LocalInvocationID.x/K_QUANTS_PER_ITERATION;  // 0...31 or 0...16
-    const uint ix  = gl_LocalInvocationID.x%K_QUANTS_PER_ITERATION;  // 0 or 0, 1
+    uint it_size = gl_WorkGroupSize.x/16;
+
+    const uint tid = gl_LocalInvocationID.x;
+    const uint itid = tid/it_size;  // 0...16
+    const uint ix  = tid%it_size;
```

Comment:
If we ever do want to run this shader with a workgroup larger than a subgroup, then this makes it so each subgroup will have the whole range of indices, spreading out the loads in a way that's likely to be bad for the memory system. So it may be nice to pack similar indices into the same subgroup. But I'm OK if you leave this as-is because I don't think we're likely to use larger workgroups, or we could revisit it later if needed.